flyway-core/src/main/java/org/flywaydb/core/internal/dbsupport/DbSupport.java

```diff
@@ -144,6 +143,30 @@ public void restoreCurrentSchema() {
     protected abstract void doChangeCurrentSchemaTo(String schema) throws SQLException;
 
     /**
+     * Retrieves the current user.
+     *
+     * @return The current user for this connection.
+     */
+    public String getCurrentUserName() {
```

Comment:
Many databases differentiate Session User, Current User (user name of current execution context) and Current Schema. Among them are Oracle, PostgreSQL, MS SQL Server and maybe others. This is a part of the SQL standard, so it may be useful for any database.

But I agree that putting this method to DbSupport class with a default implementation via JDBC MetaData may be erroneus at this moment. Let me keep it in OracleDbSupport and let's consider adding these methods in future, when implementations for all other databases are ready and tested.

Will fix it today.